quarkchain/evm/messages.py

```diff
@@ -560,8 +560,7 @@ def _apply_msg(ext, msg, code):
                 have=ext.get_balance(msg.sender, token_id=msg.transfer_token_id),
                 want=msg.value,
             )
-            # TODO: Why return success if the transfer failed?
-            return 1, msg.gas, []
+            return 0, 0, []
```

Comment:
May we double check how goetherum works on this to make sure the behavior is consistent?  In addition, I am wondering if there is an existing test in eth repo to test it?

Reply:
as mentioned in my first comment, in normal cases this should be dead code. however in our system it becomes possible in transfer_mnt precompiled contract (or if we have a bug on `apply_xshard_deposit`)